### PR TITLE
ENYO-5200: Enact docs: prepare 1.x to 2.0 migration guide

### PR DESCRIPTION
**Pull request by sugardave**
### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Added a migration guide for Enact 1.x to 2.0.

### Links
[//]: # (Related issues, references)
ENYO-5200

Enact-DCO-1.0-Signed-off-by: Dave Freeman dave.freeman@lge.com

### CODE REVIEW

File: docs/migration/migrating-enact-apps.md [hunk 1]
```diff
@@ -0,0 +1,531 @@
...
+```
+
+### ui
+
+#### `Group`
+`Group` now requires a unique key for `Object` type data. [Read about keys](https://reactjs.org/docs/lists-and-keys.html#keys)
+for more information.
+
+#### `Holdable`
+This component has been replaced by `ui/Touchable`.
```
Comment (ryanjduffy) [inline docs/migration/migrating-enact-apps.md]: Can we include links to the component docs here?

File: docs/migration/migrating-enact-apps.md [hunk 2]
```diff
@@ -0,0 +1,531 @@
...
+
+#### `Pressable`
+This component has been replaced by `ui/Touchable`.
```
Comment (ryanjduffy) [inline docs/migration/migrating-enact-apps.md]: Links to component docs?

File: docs/migration/migrating-enact-apps.md [hunk 3]
```diff
@@ -0,0 +1,531 @@
...
+
+#### `Repeater`
+`Repeater` now requires a unique key for `Object` type data. [Read about keys](https://reactjs.org/docs/lists-and-keys.html#keys)
+for more information.
+
+#### `Toggleable`
+The default `prop` for the default HOC configuration is changed from `'active'` to `'selected'`.
```
Comment (ryanjduffy) [inline docs/migration/migrating-enact-apps.md]: Let's add an example here:

1.x => `Touchable(Component)`, `Touchable({prop: 'selected'}, Component)`
2.x => `Touchable({prop: 'active'}, Component)`, `Touchable(Component)`

File: docs/migration/migrating-enact-apps.md [hunk 4]
```diff
@@ -0,0 +1,531 @@
...
+#### `Transition`
+`children` is now a required prop for this component.
+
+The `clipHeight` prop has been removed.  The base component's `clipHeight` is now automatically computed base on the
+`Transition`'s initial height.
+
+### webos
+
+#### `LS2Request`
+Import change.
```
Comment (ryanjduffy) [inline docs/migration/migrating-enact-apps.md]: More detail is required here. I don't remember what the change even was to suggest what detail :)

File: docs/migration/migrating-enact-apps.md [hunk 5]
```diff
@@ -0,0 +1,531 @@
...
+
+#### `VoiceReadout`
+This module (which only had exported `readAlert`) has been replaced by `webos/speech/readAlert`.
```
Comment (ryanjduffy) [inline docs/migration/migrating-enact-apps.md]: Example

1.x => `import {readAlert} from '@enact/webos/VoiceReadout';`
2.x => `import {readAlert} from '@enact/webos/speech';`

File: docs/migration/migrating-enact-apps.md [hunk 6]
```diff
@@ -0,0 +1,531 @@
...
+...
+<VirtualList
+	...
+	itemRenderer={itemRenderer}
+	...
+/>
+```
+
+### spotlight
+Selectors targeting the `spotlight` container attributes `data-container-disabled` or `data-container-muted`
```
Comment (ryanjduffy) [inline docs/migration/migrating-enact-apps.md]: Let's add a general note about adding support `spotlightId` to any `spotlight/Spottable` component.

Something like:
```
`spotlight/Spottable` added support for the `spotlightId` prop which can be used to identify the component in calls to `Spotlight.focus()`. In cases where you might have added a custom data attribute to identify a node, you can now use `spotlightId` instead.

1.x => <MyComponent data-component-id="my-first-component" />, Spotlight.focus('[data-component-id="my-first-component"]])
2.x => <MyComponent spotlightId="my-first-component" />, Spotlight.focus('my-first-component')
```

File: docs/migration/migrating-enact-apps.md [hunk 7]
```diff
@@ -0,0 +1,531 @@
...
+...
+<VirtualGridList
+	...
+	itemRenderer={itemRenderer}
+	...
+/>
+```
+
+#### `VirtualGridList.GridListImageItem`
+This component is replaced by `ui/GridListImageItem`.
```
Comment (ryanjduffy) [inline docs/migration/migrating-enact-apps.md]: Replaced by `moonstone/GridListImageItem`.

File: docs/migration/migrating-enact-apps.md [hunk 8]
```diff
@@ -0,0 +1,531 @@
...
+<VideoPlayer containerId="spotlightContainer" />
+```
+###### 2.0
+```
+<VideoPlayer spotlightId="spotlightContainer" />
+```
+The `tooltipHideDelay` prop has been removed with no replacement.
+
+#### `VirtualFlexList`
+This component is replaced by `ui/VirtualFlexList`.
```
Comment (ryanjduffy) [inline docs/migration/migrating-enact-apps.md]: Removed. Not replaced by anything afaik.

File: docs/migration/migrating-enact-apps.md [hunk 9]
```diff
@@ -0,0 +1,531 @@
...
+###### 2.0
+```
+...
+import {SliderTooltip} from '@enact/moonstone/Slider';
+...
+<Slider>
+	<SliderTooltip forceSide percent side="after" />
+</Slider>
+```
+The `onDecrement` and `onIncrement` props have been removed.  Developers can use `onKeyDown` and/or `onKeyUp`.
```
Comment (ryanjduffy) [inline docs/migration/migrating-enact-apps.md]: These event callbacks were always (mostly) internal. They were `@public` on SliderBase and IncrementSliderBase but not available on the common exports. Let's clarify that a bit here (and in IncrementSlider).

File: docs/migration/migrating-enact-apps.md [hunk 10]
```diff
@@ -0,0 +1,531 @@
...
+```
+<Slider /> // `horizontal` prop is `true` by default.
+<Slider vertical />
+```
+###### 2.0
+```
+<Slider /> // `orientation` prop is `"horizontal"` by default.
+<Slider orientation="vertical" />
+```
+The `tooltipAsPercent`, `tooltipSide`, and `tooltipForceSide` props for the built-in tooltip
```
Comment (ryanjduffy) [inline docs/migration/migrating-enact-apps.md]: `forceSide` was actually removed altogether and replaced with a generalized `side` prop that supports both locale-aware and locale-independent values.

1.x => `<Slider tooltip tooltipForceSide tooltipSide="after" orientation="vertical" />`
2.x => `<Slider orientation="vertical"><SliderTooltip side="right" /></Slider>`

File: docs/migration/migrating-enact-apps.md [hunk 11]
```diff
@@ -0,0 +1,531 @@
...
+
+#### `Input`
+The `<input>` element's styles have changed for `height`, `vertical-align`, and `margin`s.  Some previously used
+sizing and positioning CSS may no longer be necessary.  Developers should verify their layouts.
+
+#### `Item.OverlayDecorator`, `Item.Overlay`, and `Item.ItemOverlay`
+These components are all replaced by `moonstone/SlotItem`.
+
+#### `Marquee.MarqueeText`
+This component is replaced by `moonstone/Marquee`.
```
Comment (ryanjduffy) [inline docs/migration/migrating-enact-apps.md]: Add import example

1.x => `import {MarqueeText} from '@enact/moonstone/Marquee';`
2.x => `import Marquee from '@enact/moonstone/Marquee';`

File: docs/migration/migrating-enact-apps.md [hunk 12]
```diff
@@ -0,0 +1,531 @@
...
+`side`, and `forceSide` props instead.
+##### Example
+###### 1.x
+```
+<IncrementSlider tooltip tooltipAsPercent tooltipForceSide tooltipSide="after" />
+```
+###### 2.0
+```
+...
+import {IncrementSliderTooltip} from '@enact/moonstone/IncrementSlider';
```
Comment (ryanjduffy) [inline docs/migration/migrating-enact-apps.md]: include `IncrementSlider` in the import as well

File: docs/migration/migrating-enact-apps.md [hunk 13]
```diff
@@ -0,0 +1,531 @@
...
+```
+<IncrementSlider /> // `horizontal` prop is `true` by default.
+<IncrementSlider vertical />
+```
+###### 2.0
+```
+<IncrementSlider /> // `orientation` prop is `"horizontal"` by default.
+<IncrementSlider orientation="vertical" />
+```
+The `tooltipAsPercent`, `tooltipSide`, and `tooltipForceSide` props for the built-in tooltip
```
Comment (ryanjduffy) [inline docs/migration/migrating-enact-apps.md]: See comment on `Slider` about these props. Same applies here

**Comment by sugardave** [comment]
It seems I left some of my temporary notes in place :doh:

**Pull request merged**